File: LeetCode/cpp/2215_Find_the_Difference_of_Two_Arrays.cpp
class Solution {
public:
	vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
		vector<vector<int>> answer(2, vector<int>(0));
		set<int> s1, s2;

		for (int i = 0; i < nums1.size(); i++) {
			s1.insert(nums1[i]);
		}

		for (int i = 0; i < nums2.size(); i++) {
			s2.insert(nums2[i]);
		}

		for (int s : s1) {
			if (s2.find(s) == s2.end()) {
				answer[0].push_back(s);
			}
		}

		for (int s : s2) {
			if (s1.find(s) == s1.end()) {
				answer[1].push_back(s);
			}
		}

		return answer;
	}
};

File: LeetCode/java/2215_Find_the_Difference_of_Two_Arrays.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<List<Integer>> answer = new ArrayList<>();
        Set<Integer> s1 = new HashSet<>();
        Set<Integer> s2 = new HashSet<>();

        for (int i = 0; i < nums1.length; i++) {
            s1.add(nums1[i]);
        }

        for (int i = 0; i < nums2.length; i++) {
            s2.add(nums2[i]);
        }

        List<Integer> list = new ArrayList<>();
        for (int s : s1) {
            if (!s2.contains(s)) {
                list.add(s);
            }
        }

        answer.add(list);
        list = new ArrayList<>();

        for (int s : s2) {
            if (!s1.contains(s)) {
                list.add(s);
            }
        }

        answer.add(list);
        return answer;
    }
}